quantumdb-core/src/main/java/io/quantumdb/core/schema/definitions/Table.java

```diff
@@ -101,29 +101,35 @@ else if (parent != null && this.parent == null) {
 
 	public Table addIndex(Index index) {
 		checkArgument(index != null, "You must specify an 'index'.");
-		checkState(!containsIndex(index.getColumns()), "Table already contains an index: " + index.getIndexName() + " with columns: " + index.getColumns());
+		checkState(!containsIndexName(index.getIndexName()), "Table already contains an index with the name: " + index.getIndexName());
 
 		indexes.add(index);
 		index.setParent(this);
 		return this;
 	}
 
+	public boolean containsIndexName(String indexName) {
+		checkArgument(indexName != null, "The indexName cannot be null.");
+
+		return indexes.stream().anyMatch(i -> i.getIndexName().equals(indexName));
+	}
+
 	public boolean containsIndex(String... columns) {
-		return containsIndex(Sets.newHashSet(columns));
+		return containsIndex(Lists.newArrayList(columns));
 	}
 
-	public boolean containsIndex(Collection<String> columns) {
+	public boolean containsIndex(List<String> columns) {
 		checkArgument(!columns.isEmpty(), "You must specify at least one entry in 'columns'.");
 
 		return indexes.stream()
-				.anyMatch(c -> Sets.newHashSet(c.getColumns()).equals(Sets.newHashSet(columns)));
+				.anyMatch(i -> i.getColumns().equals(columns));
```

Comment:
I made a mistake in believing column ordering in indexes are not important, it is, so everything is converted to a list.